packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/exceptions/SyncExceptions.kt

```diff
@@ -44,22 +51,56 @@ public open class SyncException : AppException {
  *
  * @see io.realm.kotlin.mongodb.sync.SyncConfiguration.Builder.errorHandler
  */
-public class UnrecoverableSyncException : SyncException {
-    internal constructor(message: String) : super(message)
-}
+public class UnrecoverableSyncException internal constructor(message: String) :
+    SyncException(message)
 
 /**
  * Thrown when the type of sync used by the server does not match the one used by the client, i.e.
  * the server and client disagrees whether to use Partition-based or Flexible Sync.
  */
-public class WrongSyncTypeException : SyncException {
-    internal constructor(message: String) : super(message)
-}
+public class WrongSyncTypeException internal constructor(message: String) : SyncException(message)
 
 /**
  * Thrown when the server does not support one or more of the queries defined in the
  * [io.realm.kotlin.mongodb.sync.SubscriptionSet].
  */
-public class BadFlexibleSyncQueryException : SyncException {
-    internal constructor(message: String) : super(message)
+public class BadFlexibleSyncQueryException internal constructor(message: String) :
+    SyncException(message)
+
+/**
+ * Thrown when the server undoes one or more client writes. Details on undone writes can be found in
+ * [compensatingWrites].
+ */
+public class CompensatingWriteException internal constructor(
+    message: String,
+    compensatingWrites: Array<CoreCompensatingWriteInfo>
+) : SyncException(message) {
+    public val compensatingWrites: List<CompensatingWriteInfo> = compensatingWrites.map {
```

Comment:
Missing documentation for this property.

Reply:
Since this property is inside a class already called `CompensatingWrite*`, it seems a bit double to have the property named `compensatingWrites` as well. Not sure if just called it `writes` or `reversedWrites` are better? 🤔